sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CamelMessageTextMap.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Playtika
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.playtika.sleuth.camel;
+
+import lombok.AllArgsConstructor;
+import org.apache.camel.Message;
+import org.springframework.cloud.sleuth.SpanTextMap;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toMap;
+import static org.springframework.util.StringUtils.hasText;
+
+@AllArgsConstructor
+public class CamelMessageTextMap implements SpanTextMap {
+
+    private final Message delegate;
+
+    @Override
+    public Iterator<Map.Entry<String, String>> iterator() {
+        return delegate.getHeaders()
+                .entrySet().stream()
+                .collect(toMap(Map.Entry::getKey, entry -> String.valueOf(entry.getValue())))
+                .entrySet()
+                .iterator();
+    }
+
+    @Override
+    public void put(String key, String value) {
+        if (!hasText(value)) {
```

Comment:
No need, all headers involved in tracing should have value

Reply:
you need to write logs at least

Reply:
should I add logs here? it;s just helper class for injecting headers

Reply:
should be quite strange case to receive empty value. I know in our propagation tests.. we don't check for empty. it would hint at a strange propagation process. If being quite strict checking empty sounds reasonable, but yeah it hints at something likely worse if someone propagates an empty header. up to you